woocommerce/woocommerce
# Add onboarding/plugins REST endpoint

### Changes proposed in this Pull Request:

This PR adds new REST endpoints for onboarding plugin installation page.

We already have async install and status check endpoints in [Plugins.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Plugins.php), but we wanted to have dedicated endpoints for the onboarding experience with real-time error check.

1. Added the following endpoints.

POST /wp-json/wc-admin/onboarding/plugins/install-async
GET /wp-json/wc-admin/onboarding/plugins/scheduled-installs/:job-id

You can find the API spec [here](https://github.com/woocommerce/team-ghidorah/issues/212)

2. Updated [PluginsHelper::install](https://github.com/woocommerce/woocommerce/pull/38174/files#diff-38f216066764e2afb70406832279e29873dc34008334384bfd79011a95188bbbR155) to accept an instance of [PluginsInstallLogger](https://github.com/woocommerce/woocommerce/pull/38174/files#diff-e9b97940cda314d3b826232f50920d76b0bbb72ac9df1970e446b7de61956c02). At the moment, we only use [AsyncPluginsInstallLogger](https://github.com/woocommerce/woocommerce/pull/38174/files#diff-a0e49d943df2b73567b8052b507c6a578a06421a9e940c9e65ec207288e97991)

Closes https://github.com/woocommerce/team-ghidorah/issues/212



### How to test the changes in this Pull Request:

1. Make a request to POST `/wp-json/wc-admin/onboarding/plugins/install-async` with the following JSON request

```
{
    "plugins": ["test"]
}
```

2. The response should contain `job_id` field. Copy it.
3. Make a request to GET `/wp-json/wc-admin/onboarding/plugins/scheduled-installs/:job_id`. It should return the following response.

```
{
    "job_id": ":job_id",
    "status": "complete",
    "plugins": {
        "test": {
            "status": "failed",
            "errors": [
                "The requested plugin `test` could not be installed. Plugin API call failed."
            ],
            "install_duration": 0
        }
    }
}
```

4. Repeat steps 1 ~ 3, but with a valid plug-in name this time.
